src/client/mod.rs

```diff
@@ -290,464 +215,122 @@ impl Client {
         Ok(Client {
             flags: flags,
             cluster_id: cluster_id,
-            mmv_path: mmv_path,
-            wi: MMVWriterInfo::new()
+            mmv_path: mmv_path
         })
     }
+    
+    pub fn export(&self, metrics: &mut [&mut MMVWriter]) -> io::Result<()> {
```

Comment:
> exactly, and since we are not establishing a 1:1 relationship between a metric and a client that reports it, we should probably avoid mutability

mutability in this case is necessary because export is modifying the `mmap_view` field of every metric. That wouldn't be possible if metrics were passed as immutable refs.

> we probably don't want the list of metrics exported to be mutable

`iter_mut()`, used [here](https://github.com/performancecopilot/hornet/pull/27/files#diff-77939fc9650773355cff92fe8aef3e90R326), requires that the slice itself be mutable.